GitCommands/PathUtil.cs

```diff
@@ -72,20 +72,32 @@ public static bool IsLocalFile(string fileName)
         }
 
         /// <summary>
-        /// A naive way to check whether the given path is a URL by checking
-        /// whether it starts with either 'http', 'ssh' or 'git'.
+        /// A way to check whether the given path contains an URL by checking
+        /// whether it can be parsed by the .Net framework URI parsers.
         /// </summary>
+        /// <remarks>
+        /// Uri.IsWellFormedUriString is used instead of Uri.TryCreate as file URIs
+        /// of the form X:\\directory\\filename should not be treated as URLs (doing so causes
+        /// LocalRepositoryManager.AddAsMostRecentAsync to throw a NotSupportedException)
+        /// </remarks>
         /// <param name="path">A path to check.</param>
-        /// <returns><see langword="true"/> if the given path starts with 'http', 'ssh' or 'git'; otherwise <see langword="false"/>.</returns>
+        /// <returns><see langword="true"/> if the given path contains a URL; otherwise <see langword="false"/>.</returns>
         [Pure]
         public static bool IsUrl(string path)
         {
-            return !string.IsNullOrEmpty(path)
-                && (path.StartsWith("http:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("https:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("git:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("ssh:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("file:", StringComparison.CurrentCultureIgnoreCase));
+            if (string.IsNullOrEmpty(path))
+            {
+                return false;
+            }
+
+            string[] pathElements = path.Split(' ');
+            bool validUrl = false;
+            foreach (string str in pathElements)
+            {
+                validUrl |= Uri.IsWellFormedUriString(str, UriKind.Absolute);
+            }
```

Comment:
Likewise this method's purpose is to validate the input string, and not break it apart and check each part. The proposal makes the behaviour surprising - as now the question "Is 'git clone https://github.com/gitextensions/gitextensions' a URL?" results in "Yes", which is unexpected to anyone looking at the signature.

Reply:
Agree. If the caller wants to check individual words, they can split the string and call this on each.